### PR TITLE
[firebase_analytics] Wire the platform interface.

### PR DESCRIPTION
**Pull request by tugorez**
## Description

Use the platform interface introduced in https://github.com/FirebaseExtended/flutterfire/pull/1529.

## Related Issues

https://github.com/flutter/flutter/issues/45289

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart [hunk 1]
```diff
@@ -869,17 +858,14 @@ class FirebaseAnalytics {
...
   /// Sets the duration of inactivity that terminates the current session.
   ///
   /// The default value is 1800000 (30 minutes).
   Future<void> setSessionTimeoutDuration(int milliseconds) async {
     if (milliseconds == null) {
       throw ArgumentError.notNull('milliseconds');
     }
-    await _channel.invokeMethod<void>(
-        'setSessionTimeoutDuration', milliseconds);
+    await setSessionTimeoutDuration(milliseconds);
```
Comment (ditman) [inline packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart:868]: `_platformInstance.setSessionTimeoutDuration(milliseconds)`?

File: packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart [hunk 2]
```diff
@@ -3,18 +3,14 @@
...
 import 'dart:async';
-
 import 'package:flutter/foundation.dart';
-import 'package:flutter/services.dart';
 import 'package:meta/meta.dart';
+import 'package:firebase_analytics_platform_interface/firebase_analytics_platform_interface.dart';
 
-const MethodChannel firebaseChannel =
-    MethodChannel('plugins.flutter.io/firebase_analytics');
+final _platformInstance = FirebaseAnalyticsPlatform.instance;
```
Comment (ditman) [inline packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart:10]: Move this inside the FirebaseAnalytics class (so it's a final instance variable and not a global...)

Comment (ditman) [reply packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart:10]: (Then modify the constructor of FirebaseAnalyticsAndroid to receive a _platformInstance as a parameter, so you can pass the instance you already have into the FirebaseAnalyticsAndroid object)

File: packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart (hunk 1 above)
Comment (tugorez) [reply packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart:868]: Nice catch. Thanks :)

File: packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart (hunk 2 above)
Comment (tugorez) [reply packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart:10]: Done!

**Comment by tugorez** [comment]
> This looks good but please, rebase your branch. Somebody pushed a bunch of version bumps, and now your changes are conflicting.
> 
> Also, ppplease update the "Description" paragraph of this PR with what it does, for others to understand what's going on :P (and ourselves in the future!)
> 
> Other than that, LGTM!

Thansk for the review @ditman. Done!

**Comment by ditman** [comment]
Hey @kroikie, can you PTAL? This is a required step to bringing firebase_analytics to the web. Thanks!

**Comment by ditman** [comment]
Hey @Ehesp, can you PTAL? This is a required step to bringing firebase_analytics to the web [also in PR](flutter/flutter#45289). Thanks!

**Pull request merged**